build_scripts/build_utils.py

```diff
@@ -58,6 +63,27 @@ def _clean_out_dir(msg):
       os.remove(path)
 
 
+def get_module_and_log_file_from_result_msg(msg):
+  if msg.startswith(FAILURE_MSG_PREFIX):
+    _, mod_name, log_file = msg.split(RESULT_MSG_SEP)
```

Comment:
Here you assume that the msg always has three components, but in failure_msg, adding the log file is optional.

Reply:
Good catch and thanks for it. It showed me that a mere copy paste from the other file is not good enough. I restructured the code a bit to make it cleaner and fix the problem you have pointed out.